model/model.go

```diff
@@ -764,6 +769,54 @@ func (pi *PartitionInfo) GetNameByID(id int64) string {
 	return ""
 }
 
+func (pi *PartitionInfo) GetStateByID(id int64) SchemaState {
+	for _, pstate := range pi.States {
+		if pstate.ID == id {
+			return pstate.State
+		}
+	}
+	return StatePublic
+}
+
+func (pi *PartitionInfo) SetStateByID(id int64, state SchemaState) {
+	newState := PartitionState{ID: id, State: state}
+	for i, pstate := range pi.States {
+		if pstate.ID == id {
+			pi.States[i] = newState
+			return
+		}
+	}
+	if pi.States == nil {
+		pi.States = make([]PartitionState, 0, 1)
+	}
+	pi.States = append(pi.States, newState)
+}
+
+func (pi *PartitionInfo) GCPartitionStates() {
```

Comment:
When will you call it?

Reply:
Haven't decided yet. Maybe when a partition is being dropped, we can call this function to gc the unmaintained parition states.